modules/DragonridingVigor.lua

```diff
@@ -76,47 +222,115 @@ function DragonridingVigor.prototype:CheckShouldShow(event, unit, info)
 		if self.moduleSettings.hideBlizz then
 			self:ShowBlizz()
 		end
-
 		return
 	end
 
 	self:Show(true)
-
-	-- if knowsAlternateMountEnum and UnitPowerMax(self.unit, unitPowerType) > 0 then
-	-- 	self:Show(true)
-	-- elseif not knowsAlternateMountEnum and IceHUD:HasAnyBuff("player", DragonridingBuffs) then
-	-- 	self:Show(true)
-	-- else
-	-- 	self:Show(false)
-	-- 	if self.moduleSettings.hideBlizz then
-	-- 		self:ShowBlizz()
-	-- 	end
-	-- end
 end
 
 function DragonridingVigor.prototype:UpdateRunePower(event, arg1, arg2)
 	self:UpdateVigorRecharge("internal")
+
+	-- In charge mode, the base class still reads UnitPower/UnitPowerMax.
+	-- We temporarily shim those APIs so IceClassPowerCounter renders spell charges as "power".
```

Comment:
This makes me super nervous. We have control over the base class, so why not add this functionality there so we don't have to shim global methods? Blizzard has been known to have code like this trip their pollution-detection system.